### ISSUE TITLE
Enhancement: Add flipRTL attribute or similar to Slider to enable automated mirroring in RTL languages

### ISSUE THREAD

**Issue by pllcoster** [Esri/calcite-design-system#9683]
**monday.com sync:** #18048703305

### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

Many calcite components appear to detect and flip direction when a parent element has the `dir="rtl"` attribute, however this is not the case for the slider component:

![image](https://github.com/Esri/calcite-design-system/assets/146282673/bfefc861-7380-4f83-ac70-cca0a8c2e78a)

According to the Material Design Docs (https://m2.material.io/design/usability/bidirectionality.html#mirroring-elements), slider elements should be mirrored in RTL languages.

Requested functionality is to have a `flipRTL` attribute added to the slider that automatically mirrors the Slider element appropriately when contained in an RTL element. A similar effect is currently possible to a achieve in a more manual fashion by setting the `mirrored` property based on the currently language direction.

### Acceptance Criteria

Addition of a `flipRTL` attribute on the Slider component, which when set, mirrors the element automatically when its direction is "rtl".

### Relevant Info

There is a separate ticket regarding some issues with Slider in RTL here (https://github.com/Esri/calcite-design-system/issues/3167) that seems to suggest that in the past, the mirroring occurred automatically in an older calcite version.

### Which Component

Slider

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Knowledge

**Comment by geospatialem** [comment]
@annierm18 `slider` will soon also feature a vertical layout via #5522, should the request above adding the ability to mirror accommodate both `layout`'s - horizontal and vertical?

**Comment by geospatialem** [comment]
> @annierm18 `slider` will soon also feature a vertical layout via #5522, should the request above adding the ability to mirror accommodate both `layout`'s - horizontal and vertical?

The i18n team recommends the above accommodate both `layout`'s `"horizontal"` and `"vertical"` - updated the estimate to accommodate both `layout` types. Also is blocked by the effort of #5522.
